framework/doc/content/syntax/FVInterfaceKernels/index.md

```diff
@@ -0,0 +1,51 @@
+# FVInterfaceKernels System
+
+For an overview of MOOSE FV please see [/fv_design.md].
+
+`FVInterfaceKernels` are meant to communicate data between at interfaces between
+subdomains. An `FVInterfaceKernel` may contribute to the residuals/Jacobians of
+a single variable, specified with the parameter `variable1`, or to multiple
+variables by also using the `variable2` parameter. There are two additional
+critical/required parameters: `subdomain1` and `subdomain2`. In cases whether an
+`FVInterfaceKernel` is operating on two variables, `subdomain1` should
+correspond to the subdomain(s) neighboring the `boundary` parameter that
+`variable1` lives on, and similarly for `subdomain2` and `variable2`. By
```

Comment:
can we have this checking be done by the interface kernel at construction time

Reply:
I'm not quite sure what you mean. Whether '1' is on the elem or neighbor side can be different depending on the `FaceInfo` object. `elem` and `neighbor` determination for `FaceInfo` is determined solely by element numbering (well not entirely true in the case of adaptivity) and is out of a user's control.

Reply:
I meant maybe we could check for variable1 to exist on subdomain1 and variable2 to exist on subdomain2

Reply:
`FVInterrfaceKernel` developers could create something like a
```c++
std::unordered_map<const FaceInfo *, std::pair<const ADMaterialProperty *, const ADMaterialProperty *>> _fi_to_elem_neighbor_coeffs;
```
during `initialSetup` and after mesh changes if they wanted to avoid determining 1,2 -> fi_elem, fi_neighbor every time we evaluate a residual/Jacobian...

Reply:
The fundamental issue here is that `FaceInfo` objects do not have sidedness so there is this ambiguity that doesn't exist for residual compute objects associated with sidesets, which are associated with distinct element faces.

Reply:
Error checking added